### PR TITLE
PLAT-52825: Refactoring to reduce ref usage in VirtualList, Scrollable

### PR DESCRIPTION
**Pull request by ybsung**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

Sometimes we referred the child or the grandchild's props even though those props could be referred from the current component.

e.g
AS-IS
```
const {direction} = this.uiRef.props;
```
TO-BE
```
const {direction} = this.props;
```

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

Refer the props efficiently between Moonstone and UI libraries and between Scrollable and VirtualList from this.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

Renamed the following variables
```
childComponentProps as childUiComponentProps
initContainerRef as initUiContainerRef
initItemContainerRef as initUiItemContainerRef
initChildRef as initUiChildRef
virtualListProps as scrollableProps
```

### Links
[//]: # (Related issues, references)

PLAT-52825

### Comments

Enact-DCO-1.0-Signed-off-by: YB Sung (yb.sung@lge.com)

### CODE REVIEW

File: packages/ui/VirtualList/VirtualListBase.js [hunk 1]
```diff
@@ -267,7 +267,7 @@ const VirtualListBaseFactory = (type) => {
 		threshold = 0
 		maxFirstIndex = 0
 		prevFirstIndex = 0
-		curDataSize = 0
+		prevDataSize = 0
```
Comment (MikyungKim) [inline packages/ui/VirtualList/VirtualListBase.js]: I think `curDataSize` makes more sense.

File: packages/moonstone/Scrollable/Scrollable.js [hunk 2]
```diff
@@ -91,6 +91,19 @@ class Scrollable extends Component {
...
+		 *
+		 * Valid values are:
+		 * * `'horizontal'`, and
+		 * * `'vertical'`.
+		 *
+		 * @type {String}
+		 * @default 'vertical'
+		 * @public
+		 */
+		direction: PropTypes.oneOf(['horizontal', 'vertical']),
```
Comment (MikyungKim) [inline packages/moonstone/Scrollable/Scrollable.js]: We should add `both` case for scroller and make it private.

File: packages/moonstone/Scrollable/Scrollable.js [hunk 3]
```diff
@@ -103,6 +116,7 @@ class Scrollable extends Component {
 	}
 
 	static defaultProps = {
+		direction: 'vertical',
```
Comment (MikyungKim) [inline packages/moonstone/Scrollable/Scrollable.js]: Please remove default Props.

File: packages/ui/Scrollable/Scrollable.js [hunk 4]
```diff
@@ -108,6 +108,19 @@ class ScrollableBase extends Component {
...
+		 *
+		 * Valid values are:
+		 * * `'horizontal'`, and
+		 * * `'vertical'`.
+		 *
+		 * @type {String}
+		 * @default 'vertical'
+		 * @public
+		 */
+		direction: PropTypes.oneOf(['horizontal', 'vertical']),
```
Comment (MikyungKim) [inline packages/ui/Scrollable/Scrollable.js]: This one also needs to be private and add 'both'.

File: packages/moonstone/Scrollable/Scrollable.js [hunk 5]
```diff
@@ -447,13 +462,13 @@ class Scrollable extends Component {
...
 				containerRenderer={({ // eslint-disable-line react/jsx-no-bind
-					childComponentProps,
+					childUiComponentProps,
 					className,
 					componentCss,
 					handleScroll,
 					horizontalScrollbarProps,
-					initContainerRef,
 					initUiChildRef,
+					initUiContainerRef,
```
Comment (MikyungKim) [inline packages/moonstone/Scrollable/Scrollable.js]: Let's provide functions without `Ui` and rename it when passing it.

File: packages/moonstone/Scrollable/Scrollable.js [hunk 6]
```diff
@@ -463,14 +478,14 @@ class Scrollable extends Component {
...
-						containerRef={initContainerRef}
+						containerRef={initUiContainerRef}
 						focusableScrollbar={focusableScrollbar}
 						style={style}
 					>
 						<div className={componentCss.container}>
 							<TouchableDiv {...touchableProps}>
 								{childRenderer({
-									...childComponentProps,
+									...childUiComponentProps,
```
Comment (MikyungKim) [inline packages/moonstone/Scrollable/Scrollable.js]: I think `childComponentProps` makes more sense.

File: packages/ui/VirtualList/VirtualListBase.js [hunk 7]
```diff
@@ -714,7 +714,7 @@ const VirtualListBaseFactory = (type) => {
 			}
 		}
 
-		initItemContainerRef = (ref) => {
+		initUiItemContainerRef = (ref) => {
```
Comment (MikyungKim) [inline packages/ui/VirtualList/VirtualListBase.js]: Please remove `Ui`

File: packages/ui/VirtualList/VirtualListBase.js [hunk 8]
```diff
@@ -806,13 +806,13 @@ const ScrollableVirtualList = (props) => (
 const ScrollableVirtualListNative = (props) => (
 	<ScrollableNative
 		{...props}
-		childRenderer={({initUiChildRef, ...virtualListProps}) => ( // eslint-disable-line react/jsx-no-bind
+		childRenderer={({initChildRef, ...scrollableProps}) => ( // eslint-disable-line react/jsx-no-bind
```
Comment (MikyungKim) [inline packages/ui/VirtualList/VirtualListBase.js]: Please replace `scrollableProps` to `rest`.

File: packages/ui/Scroller/Scroller.js [hunk 9]
```diff
@@ -229,8 +229,8 @@ const Scroller = (props) => (
 const ScrollerNative = (props) => (
 	<ScrollableNative
 		{...props}
-		childRenderer={({initUiChildRef, ...scrollerProps}) => ( // eslint-disable-line react/jsx-no-bind
-			<ScrollerBase {...scrollerProps} ref={initUiChildRef} />
+		childRenderer={({initChildRef, ...scrollerProps}) => ( // eslint-disable-line react/jsx-no-bind
```
Comment (MikyungKim) [inline packages/ui/Scroller/Scroller.js]: Please replace to `scrollerProps` to `rest`.

**Comment by MikyungKim** [comment]
LGTM

File: packages/moonstone/Scrollable/Scrollable.js [hunk 10]
```diff
@@ -178,7 +192,8 @@ class Scrollable extends Component {
 
 	onFocus = (ev) => {
 		const
-			{isDragging, animator, direction} = this.uiRef,
+			{direction} = this.props,
+			{isDragging, animator} = this.uiRef,
```
Comment (0x64) [inline packages/moonstone/Scrollable/Scrollable.js]: Needs to re-order names?

**Pull request merged**